### PR TITLE
[NVIDIA] Fix CUDA version checks

### PR DESCRIPTION
**Pull request by kaixih**
This PR allows the `is_cuda_compute_capability_at_least` to support newer GPU compute capability like 10.0.

### CODE REVIEW

**Comment by kaixih** [comment]
@hawkinsp Can you review or assign?

File: jax/_src/test_util.py
Comment (hawkinsp) [inline jax/_src/test_util.py:496]: We should avoid floating point comparisons where we aren't dealing with approximate values. Here I suggest comparing as tuples?

If you form `tuple(int(x) for x in capability.split("."))` then those tuples should compare as you would expect versions to compare (lexicographically).

Comment (kaixih) [reply jax/_src/test_util.py]: Thanks for the suggestion. Done. PTAL.

**Comment by kaixih** [comment]
Gentle ping @hawkinsp

**Comment by kaixih** [comment]
@hawkinsp Do I need to do anything to unblock this? Or is it just waiting for approval from your side? Thx.

**Comment by hawkinsp** [comment]
I'm OOO this week, maybe @dfm could take a look?

**Comment by dfm** [comment]
Thanks @kaixih! This looks good to me. Can you squash the commits into one?

**Comment by kaixih** [comment]
Sure. Done. PTAL. @dfm

**Review by dfm — APPROVED** [review]
Thanks!

**Pull request merged**
